docs/products/m3db/concepts/scaling-m3.rst

```diff
@@ -0,0 +1,162 @@
+
+Aiven for M3 has hardcoded limit for total number of files, which limits the total number of block shards that can be allocated for single node. If desired namespace configuration would exceed this limit, by increasing number of nodes, it is possible to have more total blocks in the cluster.
+
+CPU scaling
+-----------
+
+In general CPU usage scales with the amount of work done; the number of operations (especially if having to open new SSL connections) contributes significantly, as does the data mass described above. In general:
+
+* M3Aggregators are not particularly CPU bound.
+
+* M3Coordinator's main resource that is used is CPU. However, if there is some sort of network congestion and requests pile up, the memory usage can also spike. 
+
+Memory scaling
+--------------
+
+In our experience memory has been the hardest to scale in M3 clusters. The memory usage of M3 consists of:
+
+* fixed base cost (pretty low, ~hundreds of megabytes if configured with small or no pools)
+
+* per-request overhead (main cost in M3Coordinator)
+
+* cached data for queries
+
+* storing data being aggregated in memory; only applicable if aggregated namespaces are configured (main cost in M3Aggregator or significant one in M3Coordinator if M3Aggregator is not in use)
+
+* storing data not yet written to disk - current block (main cost in M3DB)
+
+The first three are not that interesting as they do not scale directly with configuration and-or points stored. However, the last two are usually the ones that cause scalability problems.
+
+The data being stored for aggregation scales with number of namespaces, times number of time series, and also number of points fitting in a single block for that namespace. Due to that, each added namespace adds significant memory cost (unless some of the time series are filtered).
+
+The data in current blocks (not yet persisted to disk as filesets) has also same pattern. It grows as any of the following grow:
+
+* number of time series
+
+* number of points within single block
+
+* number of configured namespaces
+
+For efficient on-disk storage, M3's default recommendation is to have about 720 points per block. So for 15 seconds per point case, that would indicate 180 minutes, or 3 hours per block.  Following the recommendation may be worth it for high resolution namespaces, but for longer lived ones it may lead to too much data being kept in memory. For example, if there is a point per day, closing block every two years would not work. In that sort of case, e.g. day(s) or at most week's block size duration is recommended.
+
+The data within current block (and for a bit longer depending on buffer past duration) is stored in RAM until they are flushed to disk, and this is multiplied by number of time series AND it is also multiplied by number of namespaces that contain those particular points. 
```

Comment:
What is buffer past duration? Is it necessary to specify it now?

Reply:
"this is multiplied by number of time series AND it is also multiplied by number of namespaces that contain those particular points"
->
"this is multiplied by number of time series AND by the number of namespaces containing those particular points"